packages/frontend/src/state/controller/hooks.ts

```diff
@@ -360,36 +359,30 @@ const useNormFactor = () => {
   return normFactor
 }
 
-const useIndex = () => {
-  const address = useAtomValue(addressAtom)
-  const web3 = useAtomValue(web3Atom)
-  const networkId = useAtomValue(networkIdAtom)
+export const useIndex = () => {
+  const { getTwapSafe } = useOracle()
   const [index, setIndex] = useAtom(indexAtom)
-  const contract = useAtomValue(controllerContractAtom)
+  const { ethUsdcPool, weth, usdc } = useAtomValue(addressesAtom)
 
-  useEffect(() => {
-    if (!contract) return
-    getIndex(1, contract).then(setIndex)
-  }, [address, networkId, contract, setIndex])
+  useAppEffect(() => {
+    const getTwapEth = () => {
+      if (ethUsdcPool && weth && usdc) {
+        getTwapSafe(ethUsdcPool, weth, usdc, 1).then((quote) => {
+          setIndex(quote)
+        })
+      }
+    }
 
-  // setup index listender
-  useEffect(() => {
-    if (!web3) return
-    const sub = web3.eth.subscribe(
-      'logs',
-      {
-        address: [ETH_USDC_POOL[networkId]],
-        topics: [SWAP_EVENT_TOPIC],
-      },
-      () => {
-        getIndex(1, contract).then(setIndex)
-      },
-    )
+    getTwapEth()
```

Comment:
Can we change this to a hook. so it will be easy to adapt for any other states. https://overreacted.io/making-setinterval-declarative-with-react-hooks/